### PR TITLE
APIS-2225

### PR DESCRIPTION
**Pull request by peteslater-ee**
Added RAML documentation for the test support service.

### CODE REVIEW

File: app/controllers/MarriageAllowanceEligibility.scala [hunk 1]
```diff
@@ -17,22 +17,28 @@
...
+  def fetch(utr: String, nino: String, firstname: String, surname: String, dateOfBirth: String, taxYearStart: String) = Action async {
     implicit request =>
       val eligibleDataSetPath = "/resources/marriage-allowance-eligibility/happy_path_eligible.json"
       val ineligibleDataSetPath = "/resources/marriage-allowance-eligibility/happy_path_ineligible.json"
 
-      nino match {
-        case "AA000003D" => Future(jsonResourceAsResponse(eligibleDataSetPath))
-        case "AA000004C" => Future(jsonResourceAsResponse(ineligibleDataSetPath))
+      utr match {
+        case "1111111111" => Future(jsonResourceAsResponse(eligibleDataSetPath))
```
Comment (AndySpaven) [inline app/controllers/MarriageAllowanceEligibility.scala]: Really 1's and 2's ?

File: app/controllers/MarriageAllowanceEligibility.scala [hunk 2]
```diff
@@ -17,22 +17,28 @@
...
+        case "2222222222" => Future(jsonResourceAsResponse(ineligibleDataSetPath))
         case _ => Future.failed(new Exception(s"National Insurance Number $nino is not supported by this API"))
       }
   }
+
+  def create(utr: SaUtr, taxYear: TaxYear) = Action async {
+    Future(Created)
```
Comment (AndySpaven) [inline app/controllers/MarriageAllowanceEligibility.scala]: I think Future(xxx) should be more correctly Future.successful(xxx).  Not sure why.

File: app/controllers/MarriageAllowanceEligibility.scala [hunk 3]
```diff
@@ -17,22 +17,28 @@
...
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-class MarriageAllowanceEligibility extends BaseController with StubResource {
-  def fetch(nino: String, firstname: String, surname: String, dateOfBirth: String, taxYearStart: String) = Action.async {
+final class MarriageAllowanceEligibility extends BaseController with StubResource {
+  def fetch(utr: String, nino: String, firstname: String, surname: String, dateOfBirth: String, taxYearStart: String) = Action async {
     implicit request =>
       val eligibleDataSetPath = "/resources/marriage-allowance-eligibility/happy_path_eligible.json"
```
Comment (AndySpaven) [inline app/controllers/MarriageAllowanceEligibility.scala]: Should these constants not be in a higher scope outside the Action (or even the func maybe)

File: app/controllers/MarriageAllowanceStatus.scala [hunk 4]
```diff
@@ -29,4 +31,8 @@ class MarriageAllowanceStatus extends BaseController with StubResource {
       val dataSetPath = "/resources/marriage-allowance-status/happy_path.json"
       Future(jsonResourceAsResponse(dataSetPath))
   }
+
+  def create(utr: SaUtr, taxYear: TaxYear) = Action async {
+    Future(Created)
```
Comment (AndySpaven) [inline app/controllers/MarriageAllowanceStatus.scala]: .successful ?

File: app/models/TaxYear.scala [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ * limitations under the License.
+ */
+
+package models
+
+import scala.util.matching.Regex
+import scala.util.matching.Regex.Match
+
+case class TaxYear(ty: String) {
+  if(!TaxYear.isValid(ty)) throw new IllegalArgumentException
```
Comment (AndySpaven) [inline app/models/TaxYear.scala]: Use require() maybe

File: app/controllers/MarriageAllowanceEligibility.scala (hunk 1 above)
Comment (peteslater-ee) [inline app/controllers/MarriageAllowanceEligibility.scala]: Yeah that's in the spec

Comment (AndySpaven) [inline app/controllers/MarriageAllowanceEligibility.scala]: Just wanted to be sure - the original values looked more real :-)

File: app/controllers/MarriageAllowanceEligibility.scala (hunk 2 above)
Comment (peteslater-ee) [inline app/controllers/MarriageAllowanceEligibility.scala]: Hmmm... nice. Without .successful a new async future is created in apply which returns with the result you pass. Waste of cycles.

File: app/controllers/MarriageAllowanceStatus.scala (hunk 4 above)
Comment (peteslater-ee) [inline app/controllers/MarriageAllowanceStatus.scala]: :+1:

File: app/models/TaxYear.scala (hunk 5 above)
Comment (peteslater-ee) [inline app/models/TaxYear.scala]: Will do

File: app/controllers/MarriageAllowanceEligibility.scala (hunk 3 above)
Comment (peteslater-ee) [inline app/controllers/MarriageAllowanceEligibility.scala]: Nah - they'll be fine where they are. It's a pattern.

File: app/controllers/MarriageAllowanceEligibility.scala (hunk 1 above)
Comment (AndySpaven) [inline app/controllers/MarriageAllowanceEligibility.scala]: .successful

Comment (peteslater-ee) [inline app/controllers/MarriageAllowanceEligibility.scala]: They should all be I think

**Pull request merged**